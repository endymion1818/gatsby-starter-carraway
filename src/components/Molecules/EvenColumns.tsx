import React, { FC } from 'react'
import Column from '../Atoms/Column'
import Row from '../Atoms/Row'
import { flexalign, size, textalign } from '../tokens'

export interface IEvenColumnsGlobalProps {
  textAlign?: string
  bufferTop?: string
  bufferBottom?: string
  verticalAlign?: string
}

export interface IRenderContentProps extends IEvenColumnsGlobalProps {
  item: { innerContent: JSX.Element }
  index: number
}

export const renderContent: FC<IRenderContentProps> = ({
  item,
  index,
  textAlign = textalign.left,
  bufferBottom = size.zero,
  bufferTop = size.zero,
  verticalAlign = flexalign.start,
}) => {
  if (!item) {
    return null
  }
  return (
    <Column
      key={index}
      textAlign={textAlign}
      bufferBottom={bufferBottom}
      bufferTop={bufferTop}
      verticalAlign={verticalAlign}
    >
      {item.innerContent}
    </Column>
  )
}

export interface IEvenColumnsProps extends IEvenColumnsGlobalProps {
  content: Array<{ innerContent: JSX.Element }>
  index?: number
}

const EvenColumns: FC<IEvenColumnsProps> = ({
  content,
  textAlign,
  bufferBottom,
  bufferTop,
  verticalAlign,
}) => (
  <Row size={content.length}>
    {content.map((item, index) =>
      renderContent({
        item,
        index,
        textAlign,
        bufferBottom,
        bufferTop,
        verticalAlign,
      })
    )}
  </Row>
)

export default EvenColumns
