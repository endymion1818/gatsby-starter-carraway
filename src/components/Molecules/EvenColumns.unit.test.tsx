import { cleanup, render } from '@testing-library/react'
import React from 'react'
import EvenColumns from './EvenColumns'

const defaultProps = {
  content: [
    {
      innerContent: <div>test</div>,
    },
  ],
  index: 0,
  textAlign: 'left',
  bufferTop: '0rem',
  bufferBottom: '0rem',
  verticalAlign: 'top',
}

describe('<EvenColumns />', () => {
  beforeEach(cleanup)
  it('Should render', () => {
    const { container } = render(<EvenColumns {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })
})
