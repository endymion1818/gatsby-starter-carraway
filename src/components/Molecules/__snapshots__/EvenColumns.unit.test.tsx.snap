// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EvenColumns /> Should render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: top;
  -ms-flex-item-align: top;
  align-self: top;
  margin-top: 0rem;
  margin-bottom: 0rem;
  text-align: left;
}

.c1 > h1,
.c1 > h2,
.c1 > h3,
.c1 > h4,
.c1 > h5,
.c1 > h6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 > p {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c1 > p ~ div[class*='Button'] {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
}

.c1 > img {
  width: 100%;
}

@media (min-width:) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(1,1fr);
    grid-gap: 15px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c0 {
    display: block !important;
  }

  .c0 > * {
    display: block !important;
    float: left;
    width: 100%;
  }

  .c0 > *:not(:first-child) {
    margin-left: 1rem;
  }

  .c0 > *:not(:last-child) {
    margin-right: 1rem;
  }

  .c0:after {
    content: '';
    display: block;
    clear: both;
  }
}

<div>
  <div
    class="c0"
    size="1"
  >
    <div
      class="c1"
    >
      <div>
        test
      </div>
    </div>
  </div>
</div>
`;
